lib/holidays/finder/rules/in_region.rb

```diff
@@ -9,7 +9,15 @@ def call(requested, available)
             # When an underscore is encountered, derive the parent regions
             # symbol and check for both.
             requested = requested.collect do |r|
-              r.to_s =~ /_/ ? [r, r.to_s.gsub(/_[\w]*$/, '').to_sym] : r
+              if r.to_s =~ /_/
+                chunks = r.to_s.split('_')
+                
+                chunks.length.downto(1).map do |num|
```

Comment:
I think this is a great change! What do you think about adding a test in the specific `test_in_region.rb` file to show the failure specifically so we can make sure we don't introduce a similar bug in the future?